### PR TITLE
Add CompressetIntList, RangeList types

### PR DESCRIPTION
**Pull request by andyjgf**

### CODE REVIEW

File: Overview.bs [hunk 1]
```diff
@@ -260,6 +269,175 @@ significant bit in the byte and the bit with the largest index is the most signi
...
+zig_zag_encoded_list = [46, 40, 61, 17, 128, 2, 2, 137]
+bytes = [2E 28 3D 11 80 01 02 02 89 01]
+//       └┘ └┘ └┘ └┘ └───┘ └┘ └┘ └───┘
+```
+</div>
+
+### RangeList ### {#rangelist}
+
+A Range is a set of integers i where min <= i <= max. It is represented by
+the pair of integers [min, max], where min <= max and min >= 0.
```
Comment (garretrieger) [inline Overview.bs:419]: Suggest replacing this paragraph with:

A RangeList encodes a set of unsigned 32 bit integers. The set is encoded as a list of disjoint intervals. Each interval is represented by two integers, a start (inclusive) and end (inclusive).

File: Overview.bs [hunk 2]
```diff
@@ -83,7 +82,9 @@ should be encoded by CBOR are given in the definition of those data types.
     <th>Data Type</th><th>Description</th><th>CBOR Major Type</th>
   </tr>
   <tr>
-    <td>Integer</td><td>An integer value range [-2^64 - 1, 2^64 - 1] inclusive.&nbsp;</td><td>0 or 1</td>
+    <td>Integer</td><td>An integer value range [-2^64 - 1, 2^64 - 1] inclusive.
+    In most cases 32 bit signed values are used, with fingerprints being 64 bit unsigned.</td>
```
[the remark this answers is not in the document]
Comment (andyjgf) [reply Overview.bs]: done

File: Overview.bs [hunk 3]
```diff
@@ -93,6 +94,14 @@ should be encoded by CBOR are given in the definition of those data types.
   </tr>
 </table>
 
+### ProtcolVersion ### {#protocol-version}
```
[the remark this answers is not in the document]
Comment (andyjgf) [reply Overview.bs]: done

File: Overview.bs [hunk 4]
```diff
@@ -279,60 +457,34 @@ for a type specifies for each field:
...
+  <tr><td>0</td><td>protocol_version</td><td>ProtocolVersion (Integer)</td></tr>
+  <tr><td>1</td><td>accept_patch_format</td><td>CompressedIntList (ByteString)</td></tr>
   <tr><td>2</td><td>codepoints_have</td><td>CompressedSet</td></tr>
   <tr><td>3</td><td>codepoints_needed</td><td>CompressedSet</td></tr>
   <tr><td>4</td><td>indices_have</td><td>CompressedSet</td></tr>
   <tr><td>5</td><td>indices_needed</td><td>CompressedSet</td></tr>
-  <tr><td>6</td><td>ordering_checksum</td><td>Integer</td></tr>
-  <tr><td>7</td><td>original_font_checksum</td><td>Integer</td></tr>
-  <tr><td>8</td><td>base_checksum</td><td>Integer</td></tr>
+  <tr><td>6</td><td>ordering_checksum</td><td>Integer (64 bit unsigned)</td></tr>
```
[the remark this answers is not in the document]
Comment (andyjgf) [reply Overview.bs:479]: done

File: Overview.bs [hunk 5]
```diff
@@ -260,6 +269,175 @@ significant bit in the byte and the bit with the largest index is the most signi
...
+```
+delta_list = [23, 20, -31, -9, 64, 1, 1, -69]
+zig_zag_encoded_list = [46, 40, 61, 17, 128, 2, 2, 137]
+```
+</div>
+
+<figcaption>Psuedo-code for Zig-Zag encoding and decoding</figcaption>
+</figure>
+
+*Add reference to proto implementation*
```
[the remark this answers is not in the document]
Comment (andyjgf) [reply Overview.bs]: done

File: Overview.bs [hunk 6]
```diff
@@ -260,6 +269,175 @@ significant bit in the byte and the bit with the largest index is the most signi
...
 </div>
 
+### CompressedIntList ### {#compressedintlist}
+
+A data structure which compactly represents a list of integers. A distinction is
+made between sorted (positive, nondecreasing) and unsorted lists. Sorted lists
+allow for greater compression. Both types of lists are encoded into a ByteString
+for transport.
+
+Unsorted lists use three steps of encoding/compression: deltas, zig-zag and varint.
```
[the remark this answers is not in the document]
Comment (andyjgf) [reply Overview.bs]: done

File: Overview.bs (hunk 1 above)
Comment (andyjgf) [reply Overview.bs]: done

File: Overview.bs [hunk 7]
```diff
@@ -260,6 +269,175 @@ significant bit in the byte and the bit with the largest index is the most signi
 
 </div>
 
+### CompressedIntList ### {#compressedintlist}
```
[the remark this answers is not in the document]
Comment (andyjgf) [reply Overview.bs]: Done

File: Overview.bs [hunk 8]
```diff
@@ -260,6 +269,175 @@ significant bit in the byte and the bit with the largest index is the most signi
...
+```
+</div>
+
+<figcaption>Psuedo-code for Zig-Zag encoding and decoding</figcaption>
+</figure>
+
+*Add reference to proto implementation*
+
+
+#### Variable Length Integer (varint) Encoding #### {#compressedintlist-varint}
```
[the remark this answers is not in the document]
Comment (andyjgf) [reply Overview.bs]: done

**Review by garretrieger — APPROVED** [review]

**Pull request merged**